canopy_waf_mod.F90

```diff
@@ -102,7 +102,7 @@ SUBROUTINE CANOPY_WAF( HCM, ZTOTHC, FAFRACK, FAFRACK0, UBZREF, Z0GHCM, &
             term1 = log( LAMDARS * ( ( (FLAMEH/HCM) +  1.0 - d_h ) / zo_h ) ) - &   !numerator
                 ( 1.0 + (delta*log((1.0/delta) + 1.0)) )
             term2 = log( LAMDARS * ( ( ( (HREF/HCM) + 1.0 - d_h) )/ zo_h ) )  !denominator
-            waf   = term1 / term2
+            waf   = abs (term1 / term2) !need absolute for flameh < 1 m conditions
```

Comment:
If this necessarily the case? Doesn't seem to be.

![image](https://user-images.githubusercontent.com/15079414/186259115-8b33182d-c228-4646-9fe7-c32bf2c6126f.png)

<details><summary>Code</summary>

```python
import matplotlib.pyplot as plt
import numpy as np

lrs = 1.25  # coeff; usually this value is used
# lrs = 1.07

h = 20  # canopy height
d_h = 0.8  # zero-plane displacement height (relative)
d = h * d_h  # displacement height
H = 6  # reference height minus h
z0_h = 0.1  # z0/h ranges from 0 to 0.22 in Fig 6
z0 = z0_h * h  # roughness length

# h_flame: flame height rt. h
h_flame = np.linspace(0.01, h, 200)

delta = (1 - d_h) / (h_flame/h)

num = (
    np.log(lrs * (h_flame/h + 1 - d_h) / z0_h)
    - 1
    + delta * np.log(1/delta + 1)
)
den = np.log(lrs * (H/h + 1 - d/h) / z0_h)

waf = num/den
# waf = np.abs(waf)

fig, [ax1, ax2] = plt.subplots(2, 1, sharex=True, figsize=(4, 4))

ax1.plot(h_flame, num)
ax1.set(title=f"Numerator; denominator = {den:.3g}")

ax2.plot(h_flame, waf)
ax2.set(xlabel="$h_\mathrm{flame}$", title="WAF")
ax2.set_xlim(xmin=0)

for ax in fig.get_axes():
    ax.axvline(1, c="0.7")

fig.tight_layout()
```
</details>

The original formula doesn't have abs. Maybe negative WAF value indicates problematic input and should set to 1 or some sort of flag value instead.

Reply:
From my quick checking, it appears that flameh<1 is setting this off, but your right its sloppy.  Needed to further diagnose and fix to avoid negative WAF.

Reply:
I think this is resolved now